packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts

```diff
@@ -218,6 +222,148 @@ export interface MetricDefinition {
     readonly regex: string;
 }
 
+/**
+ * Specifies a location in a S3 Bucket.
+ *
+ * @experimental
+ */
+export interface IS3Location {
```

Comment:
Sharing `S3Location` as both the interface and the holder of the factory functions is more standard.

As is using a `bind()` pattern if possible.